lib/did_you_mean/spell_checkers/key_error_checker.rb

```diff
@@ -8,7 +8,13 @@ def initialize(key_error)
     end
 
     def corrections
-      @corrections ||= SpellChecker.new(dictionary: @keys).correct(@key).map(&:inspect)
+      @corrections ||= similar_keys + SpellChecker.new(dictionary: @keys).correct(@key).map(&:inspect)
```

Comment:
I think it makes more sense to only suggest exact matches if they are present, then fall back to the typical spell checking approach. That should by design prevent inaccurate suggestions from creeping into the final results.

```suggestion
      @corrections ||= exact_matches.empty? ? SpellChecker.new(dictionary: @keys).correct(@key).map(&:inspect) : exact_matches
```

Reply:
Good point 👍